src/ansiblelint/rules/args.py

```diff
@@ -0,0 +1,251 @@
+from ansiblelint.errors import MatchError
+from ansiblelint.file_utils import Lintable
+from ansiblelint.rules import AnsibleLintRule, RulesCollection
+from ansiblelint.text import has_jinja
+
+_logger = logging.getLogger(__name__)
+
+
+@lru_cache
+def load_module(module_name: str) -> loader.PluginLoadContext:
+    """Load plugin from module name and cache it."""
+    return loader.module_loader.find_plugin_with_context(module_name)
+
+
+class ValidationPassed(Exception):
+    """Exception to be raised when validation passes."""
+
+
+class CustomAnsibleModule(basic.AnsibleModule):  # type: ignore
+    """Mock AnsibleModule class."""
+
+    def __init__(self, *args: str, **kwargs: str) -> None:
+        """Initialize AnsibleModule mock."""
+        super().__init__(*args, **kwargs)
+        raise ValidationPassed
+
+
+class ArgsRule(AnsibleLintRule):
+    """Validating module arguments."""
+
+    id = "args"
+    severity = "HIGH"
+    description = "Check whether tasks are using correct module options."
+    tags = ["syntax", "experimental"]
+    version_added = "v6.10.0"
+    module_aliases: dict[str, str] = {"block/always/rescue": "block/always/rescue"}
+
+    def matchtask(
+        self, task: dict[str, Any], file: Lintable | None = None
+    ) -> list[MatchError]:
+        # pylint: disable=too-many-branches,too-many-locals
+        results: list[MatchError] = []
+        module_name = task["action"]["__ansible_module_original__"]
+        failed_msg = None
+
+        if module_name in self.module_aliases:
+            return []
+
+        loaded_module = load_module(module_name)
+        module_args = {
+            key: value
+            for key, value in task["action"].items()
+            if not key.startswith("__")
+        }
+
+        with mock.patch.object(
+            mock_ansible_module, "AnsibleModule", CustomAnsibleModule
+        ):
+            spec = importlib.util.spec_from_file_location(
+                "plugin", loaded_module.plugin_resolved_path
+            )
+            if spec:
+                assert spec.loader is not None
+                module = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(module)
+            else:
+                assert file is not None
+                _logger.warning(
+                    "Unable to load module %s at %s:%s for options validation",
+                    module_name,
+                    file.filename,
+                    task[LINE_NUMBER_KEY],
+                )
+                return []
+
+            try:
+                if not hasattr(module, "main"):
+                    # skip validation for module options that are implemented as action plugin
+                    # as the option values can be changed in action plugin and are not passed
+                    # through `ArgumentSpecValidator` class as in case of modules.
+                    return []
+
+                with patch.object(
```

Comment:
Hi @ganeshrn, I am looking into fixing https://github.com/ansible/ansible-lint/issues/4717 and am wondering why you opted to patch sys.argv here instead of directly setting `_ANSIBLE_ARGS`. Do you happen to remember the rationale behind it?